### PR TITLE
Allowing Dataflow users to provide additional connection configuration.

### PR DESCRIPTION
**Pull request by sduskis**
This should address the configuration requirements of issue #484 to set the datahost.

### CODE REVIEW

File: bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableConfiguration.java [hunk 1]
```diff
@@ -94,10 +105,12 @@ public CloudBigtableConfiguration build() {
    * @param zoneId The zone where the cluster is located.
    * @param clusterId The cluster ID for the cluster.
```
Comment (kevinsi4508) [inline bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableConfiguration.java]: document new parameter

File: bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableConfiguration.java [hunk 2]
```diff
@@ -146,6 +154,9 @@ public Configuration toHBaseConfig() {
     config.set(BigtableOptionsFactory.PROJECT_ID_KEY, getProjectId());
     config.set(BigtableOptionsFactory.ZONE_KEY, getZoneId());
     config.set(BigtableOptionsFactory.CLUSTER_KEY, getClusterId());
+    for (Entry<String, String> entry : additionalConfiguration.entrySet()) {
```
Comment (kevinsi4508) [inline bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableConfiguration.java]: If additionalConfiguration contains config for CLUSTER_KEY let say, it overrides "config.set(BigtableOptionsFactory.CLUSTER_KEY, getClusterId());" Please document the order.

**Comment by kevinsi4508** [comment]
What is the reason for exposing generic configuration such as Map<String, String>?

**Comment by derjust** [comment]
Just as I was wading through the code yesterday: What about `com.google.cloud.bigtable.dataflow.CloudBigtableConfiguration.toHBaseConfig()`? As it has this `addtionalProperties`, shouldn't that applied there, too?

**Comment by sduskis** [comment]
@kevinsi4508: updated as per your comments.  We're exposing a more generic construct because the likelihood of any configuration change is as likely as the datahost, which is to say not likely, but possibly needed.  PTAL

**Comment by sduskis** [comment]
@derjust: Can you clarify your question?

If you're asking if CloudBigtableConfiguration.toHBaseConfig() was updated, then the answer is: yes, it was updated.

**Comment by derjust** [comment]
@sduskis ah found it. From what I can envision as a consumer of the library I think the request from #484 is covered in a better/more flexible way :+1:

**Comment by kevinsi4508** [comment]
LGTM

**Pull request merged**